From 02f8037976773f4314721915edd98a6fd25223d6 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 17 Mar 2015 11:09:18 +0100
Subject: [PATCH] system-settings: allow no L3 configuration for
 non-ethernet/wifi devices

Lacking BOOTPROTO implies no L3 configuration

The ethernet & wifi minimal ifcfg files keep compatibility with ancient Anaconda.
However, for vlans, bonds and bridges we want to make it possible to use L2-only
configurations.
---
 system-settings/plugins/ifcfg-rh/reader.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/system-settings/plugins/ifcfg-rh/reader.c b/system-settings/plugins/ifcfg-rh/reader.c
index 11d91f6..c5a399c 100644
--- a/system-settings/plugins/ifcfg-rh/reader.c
+++ b/system-settings/plugins/ifcfg-rh/reader.c
@@ -1286,7 +1286,7 @@ make_ip4_setting (shvarFile *ifcfg,
 		char *tmp_ip4, *tmp_prefix, *tmp_netmask;
 
 		/* If there is no BOOTPROTO, no IPADDR, no PREFIX, no NETMASK, but
-		 * valid IPv6 configuration, assume that IPv4 is disabled.  Otherwise,
+		 * valid IPv6 configuration, assume that IPv4 is disabled.  For ethernet devices,
 		 * if there is no IPv6 configuration, assume DHCP is to be used.
 		 * Happens with minimal ifcfg files like the following that anaconda
 		 * sometimes used to write out:
@@ -4371,6 +4371,9 @@ connection_from_file (const char *filename,
 	if (!nm_controlled)
 		ensure_unmanaged (parsed, unmanaged);
 
+	if (strcasecmp (type, TYPE_ETHERNET) && strcasecmp (type, TYPE_WIRELESS))
+		can_disable_ip4 = TRUE;
+
 	g_free (type);
 
 	/* Don't bother reading the connection fully if it's unmanaged or ignored */
-- 
2.1.0

